extensions/wikia/Forum/tests/ForumTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+class ForumTest extends WikiaBaseTest {
+
+
+	/*
+	 * https://wikia-inc.atlassian.net/browse/SUS-156
+	 * This test check behaviour what happen if wikia props has some leftovers
+	 * in case of race condition master->slave
+	 */
+	public function testGetBoardListShouldReturnBoardsWithoutLeftoversInWikiaProps() {
+
+		$titlesFromNamespace = [ ];
+		foreach ( [ 1, 2, 3, 4 ] as $key ) {
+			$titlesFromNamespace[ $key ] = $this->getTitleMock( $key );
+		}
+		$wikiaPropsReturn = [ 1 => 4, 2 => 7, 3 => 6, 5 => 2 ];
+		$expectedBoardsTitle = [ 1, 2, 3 ];
+
+		$titleBatchMock = $this->getMock( 'TitleBatch', [ 'getById', 'getWikiaProperties' ], [ ], '', false );
```

Comment:
Mocking classes with the explicit list of methods is a bit dangerous because the methods will be mocked even if they do not exist in the target class. As long as you do not need all other methods of TitleBatch to act normally I'd recommend removing that list here.